### ISSUE TITLE
On demand deploy packages not working.

### ISSUE THREAD

**Issue by kintaro1981** [glpi-project/glpi-inventory-plugin#382]
### Describe the bug

On demand deploy packages are are not working.

### To reproduce

1. go to deploy package tab in a computer item
2. choose a package
3. click on: I'm not on this computer: wakeup from the server

### Expected behavior

I expect that:

1. the "ondemand task" is going to be listed in the Task/Groups tabs with the others tasks.
2. the deploy starts right away

Instead:
1. I can only see that the package is listed in grey "waiting" in the deploy package tab
2. if I try to click on the + to get more info I can't - firefox console say this:
```
Uncaught SyntaxError: expected expression, got ','
    x http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:188
    Be http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    append http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    html http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    K http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    html http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    loadTabContents http://glpi/front/computer.form.php?id=13718:2045
    l http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    fireWith http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    C http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    t http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
computer.form.php:21:45
    x http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:188
    Be http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    append http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    html http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    K http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    html http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    loadTabContents http://glpi/front/computer.form.php?id=13718:2045
    l http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    fireWith http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    C http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
    t http://glpi/public/lib/base.min.js?v=53489c6e5ec38eb33efa18c9adf8b3027ed40b87:199
```

3. The agent on the computer starts, it runs scheduled tasks but not the on demand ones
4. The ondemand task is not present in the Task/Group tab
5. the ondemand task is correctly added to the Tasks page (/plugins/glpiinventory/front/task.php) 
6. In the agent log I can see the other deploy tasks running ok but not the on-demand one.

### Operating system

Windows

### GLPI Agent version

1.4, Other (See additional context below)

### GLPI version

10.0.9

### GLPIInventory plugin

1.2.3

### Additional context

fusion agent 2.5.2

**Comment by kintaro1981** [comment]
am I the only one with this issue?

**Comment by kintaro1981** [comment]
anyone have any ideas how to fix this?

**Comment by cedric-anne** [comment]
We would need to see the syntax error itself to understand the issue. It may be due to some unexpected chars in some variable, that broke the JS syntax.
For instance, on Chrome, when you have an error, you can expand the stack trace and access see the related code by clicking on the link that is shown here.
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/33253653/db67cd1e-6e42-4a19-a449-63cc6167c542)

**Comment by AnthoDingo** [comment]
Hello,

Same error with same config : 

- GLPI 10.0.9
- glpiinventory 1.2.3

The entire error : 
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/4434864/186d0707-3a50-4302-8b0d-da041333730a)


The error is occurred on the underline :
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/4434864/44202e2d-bfb7-4464-a0af-00d5bfb6e9c0)

**Comment by AnthoDingo** [comment]
Below the information :

![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/4434864/27dd3bb6-8bfe-4abf-85c0-bb2a4b475344)

**Comment by stonebuzz** [comment]
Hi @AnthoDingo 

Can you check whether you have any SQL / PHP errors from GLPI logs?

Best regards

**Comment by AnthoDingo** [comment]
Hi @stonebuzz 

The log from php_error.log file.

```
[2023-08-16 08:21:52] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "last_taskjobstate" in glpi\10.0.9\plugins\glpiinventory\inc\deploypackage.class.php at line 1582
  Backtrace :
  ...\glpiinventory\inc\deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src\CommonGLPI.php:691                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax\common.tabs.php:117                           CommonGLPI::displayStandardTab()
  
[2023-08-16 08:21:52] glpiphplog.WARNING:   *** PHP Warning (2): Trying to access array offset on value of type null in glpi\10.0.9\plugins\glpiinventory\inc\deploypackage.class.php at line 1582
  Backtrace :
  ...\glpiinventory\inc\deploypackage.class.php:1097 PluginGlpiinventoryDeployPackage->showPackageForMe()
  src\CommonGLPI.php:691                             PluginGlpiinventoryDeployPackage::displayTabContentForItem()
  ajax\common.tabs.php:117                           CommonGLPI::displayStandardTab()
```

**Comment by stonebuzz** [comment]
I can't reproduce it. I don't know if you're familiar with development, but the problem must be here.

File : ```deploypackage.class.php``` 

Function : ```getPackageForMe``` (L1726)

The last lines of this function should set the ```last_taskjobstate```  key (which is missing in the rest of the process)

Can you try to debug ?

**Comment by AnthoDingo** [comment]
@stonebuzz the function getMyDepoyPackagesState return an empty array
```
$last_job_state => array(0) { }
```

The query return nothing
```php
$taskjobstates  = current($pfTaskJobState->find(
            ['plugin_glpiinventory_taskjobs_id' => $taskjobs_id,
            'agents_id'   => $agents_id],
            ['id DESC'],
            1
        ));
```

In prod, this is what I have:
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/4434864/0f79157a-2c86-42f2-a3a6-8b27aad044a4)

$taskjobs_id = 906
$agents_id = 167

On my sql instance, when I put my resquest, nothing is found : 
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/4434864/07acd518-e4a7-4389-8968-7d3aadf97d9b)

Is it possible that on the task creation, the taskjobstate insert has an error ?

**Comment by stonebuzz** [comment]
have you prepared the task ?

**Comment by AnthoDingo** [comment]
Hi @stonebuzz ,

I push the package from the Computer element, on the tab "Packages Deployment" ( http://myserver/front/computer.form.php?id=808 => tab "Packages Deployment")

**Comment by stonebuzz** [comment]
can you run "Force Start" from task and retry
![image](https://github.com/glpi-project/glpi-inventory-plugin/assets/7335054/58344316-1029-41f6-8668-a02f38b90bba)

**Comment by stonebuzz** [comment]
Can you show me related package (all tab)

Best regards

**Comment by kintaro1981** [comment]
![immagine](https://github.com/glpi-project/glpi-inventory-plugin/assets/2357176/019dda9b-e108-4f86-897d-bdb562c300b1)
![immagine](https://github.com/glpi-project/glpi-inventory-plugin/assets/2357176/225984d6-e89b-44bc-8b01-7edccb824582)
![immagine](https://github.com/glpi-project/glpi-inventory-plugin/assets/2357176/2a27335f-4a23-44af-902b-711674d238b7)

**Comment by stonebuzz** [comment]
can you try #393

**Comment by kintaro1981** [comment]
@stonebuzz in my case the issue is fixed by #393
